wx/lib/checkbox.py

```diff
@@ -0,0 +1,807 @@
+        :type `spacing`: int
+        """
+
+        self._spacing = spacing
+
+        # The spacing between the check bitmap and the text has changed,
+        # so we must recalculate our best size and refresh ourselves.
+        self.InvalidateBestSize()
+        self.Refresh()
+
+    def GetSpacing(self):
+        """ Returns the spacing between the check bitmap and the text. """
+
+        return self._spacing
+
+    def GetValue(self):
+        """
+        Returns the state of :class:`GenCheckBox`, True if checked, False
+        otherwise.
+        """
+
+        return self._checked
+
+    def IsChecked(self):
+        """
+        This is just a maybe more readable synonym for GetValue: just as the
+        latter, it returns True if the :class:`GenCheckBox` is checked and False
+        otherwise.
+        """
+
+        return self._checked
+
+    def SetValue(self, state):
+        """
+        Sets the :class:`GenCheckBox` to the given state. This does not cause a
+        ``wx.wxEVT_COMMAND_CHECKBOX_CLICKED`` event to get emitted.
+
+        :param `state`: Set the value of the checkbox. True or False.
+        :type `state`: bool
+        """
+
+        self._checked = state
+
+        # Refresh ourselves: the bitmap has changed.
+        self.Refresh()
+
+    def OnKeyUp(self, event):
+        """
+        Handles the ``wx.EVT_KEY_UP`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.KeyEvent` to be processed.
+        :type `event`: `wx.KeyEvent`
+        """
+
+        if event.GetKeyCode() == wx.WXK_SPACE:
+            # The spacebar has been pressed: toggle our state.
+            self.SendCheckBoxEvent()
+            event.Skip()
+            return
+
+        event.Skip()
+
+    def OnSetFocus(self, event):
+        """
+        Handles the ``wx.EVT_SET_FOCUS`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.FocusEvent` to be processed.
+        :type `event`: `wx.FocusEvent`
+        """
+
+        self._hasFocus = True
+
+        # We got focus, and we want a dotted rectangle to be painted
+        # around the checkbox label, so we refresh ourselves.
+        self.Refresh()
+
+    def OnKillFocus(self, event):
+        """
+        Handles the ``wx.EVT_KILL_FOCUS`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.FocusEvent` to be processed.
+        :type `event`: `wx.FocusEvent`
+        """
+
+        self._hasFocus = False
+
+        # We lost focus, and we want a dotted rectangle to be cleared
+        # around the checkbox label, so we refresh ourselves.
+        self.Refresh()
+
+    def OnPaint(self, event):
+        """
+        Handles the ``wx.EVT_PAINT`` event for :class:`GenCheckBox`.
+
+        :param `event`: A `wx.PaintEvent` to be processed.
+        :type `event`: `wx.PaintEvent`
+        """
+
+        # If you want to reduce flicker, a good starting point is to
+        # use wx.BufferedPaintDC .
+        dc = wx.BufferedPaintDC(self)
```

Comment:
`wx.AutoBufferedPaintDC` would be marginally better

Reply:
Changed to Auto and added commenting